### PR TITLE
add note about how we update the builder image

### PR DESCRIPTION
**Pull request by creviera**
## Status

Ready for review

## Description of Changes

Add a note in the docs about how we ignore the test failure that checks if the container that builds our debian packages is up-to-date with the latest packages. The test failure is expected and future improvements to this process can be made after our Focal upgrade, see  see https://github.com/freedomofpress/securedrop/issues/4533.

### CODE REVIEW

File: docs/development/release_management.rst
Comment (eloquence) [inline docs/development/release_management.rst:138]: It it necessary to create an issue to update the builder image (historically it doesn't look like we've done so), or is it standard operating procedure to always update it before building the packages for the final release?

Comment (creviera) [reply docs/development/release_management.rst:138]: I think you're right. Perhaps creating an issue to update the builder image is overkill. We could just add in the "Release Process" section below a step for updating the builder image if this test fails. So the note here could say something more like:
```
     .. note:: Since we don't update the container that builds the debian packages at runtime, you
               might end up seeing ``test_ensure_no_updates_avail`` fail in the build output. This
               tells us that our builder image needs to be updated to ensure we are using the latest
               security packages. You can ignore this test failure for release candidates, because
               it will be updated as part of the final release (linking to the release process section).
```

Comment (eloquence) [reply docs/development/release_management.rst:138]: That sounds right to me -- @emkll, does that make sense to you as well?

Comment (emkll) [reply docs/development/release_management.rst:138]: Yes, @creviera that language looks good to me. We could also link to the build instructions, since the release management documentation is generally developer-facing: docs/development/dockerbuildmaint.rst


Note that in the "Final Release" section of Release issues, we have a check box in place to ensure the builder is up-to-date, so i agree that opening an issue may be redundant here:
```
 Ensure builder in release branch is updated and/or update builder image
```

Comment (creviera) [reply docs/development/release_management.rst]: okay i also update docs/development/dockerbuildmaint.rst so this is ready again for review

File: docs/development/dockerbuildmaint.rst
Comment (eloquence) [inline docs/development/dockerbuildmaint.rst:12]: typo: bulid->build

File: docs/development/dockerbuildmaint.rst
Comment (eloquence) [inline docs/development/dockerbuildmaint.rst:36]: Could equivalent security properties be achieved by including the exact version information in the build logs automatically, as suggested in https://github.com/freedomofpress/securedrop/issues/4533#issuecomment-705717195 ? If so, it does not seem a compelling argument to include in favor of the current process.

**Comment by eloquence** [comment]
Looks good to me (one smol typo), but since I'm not directly familiar with the builder image update process, I would appreciate review from a SecureDrop maintainer before merging.

File: docs/development/dockerbuildmaint.rst
Comment (conorsch) [inline docs/development/dockerbuildmaint.rst:10]: We could make these bolded URLs actual URLs to be a bit easier on folks, but that's not a change that came in via this PR.

File: docs/development/dockerbuildmaint.rst
Comment (conorsch) [reply docs/development/dockerbuildmaint.rst:36]: The docs presented here are definitely accurate in terms of the process we actually adhere to. I remain interested in getting the SD core debs fully reproducible, so we can stop fussing with the build container so much. After Focal, that seems a reasonable goal again, consistent with discussion in https://github.com/freedomofpress/securedrop/issues/4533. 

All that to say, the "note" as written here LGTM!

File: docs/development/release_management.rst
Comment (conorsch) [inline docs/development/release_management.rst:51]: Much clearer, thank you for including the explicit URLs!

File: docs/development/release_management.rst
Comment (conorsch) [inline docs/development/release_management.rst:150]: Locally, the way I determine this is by running `git status` and checking if a file was _modified_ by my changes. If so, I run `git reset <file>` on each of them, and commit only the new packages. That's something CI could check for us, even in the dev-lfs repo, that might be a little friendlier to developers. Weighed against continued progress toward reproducibility, when we can trust a bot to file the PR, I very much prefer these docs you've got here.

**Review by conorsch — APPROVED** [review]
Looks great! Thanks for preparing, @creviera. There are a few procedures in here we'll surely revisit in the near future, but no question that the changes you've made here are more accurate, and should be included in the dev docs right away.

**Pull request merged**
